### PR TITLE
Use response of the error handler as fragment content

### LINKED ISSUE #66: Replace fragment content in error handler
Is it possible to generate some html to replace a fragment's response in the response handler, instead of the documented redirect. 

For example:

Instead of something like:
```
"statusCodeHandlers": {
   "403": {
     "fn":"handleUnauthorised"
  }
}

'handleUnauthorised"': function(req, res, variables, data, options, err) {
    res.writeHead(302, {location: '/unauthorised'});
    res.end('');
}
```

It would be something like

```
"statusCodeHandlers": {
   "403": {
     "fn":"handleUnauthorised"
  }
}

'handleUnauthorised"': function(req, res, variables, data, options, err) {
   return "<p>You're not allowed to see this fragment, but others will load</p>";
}
```

### PR DESCRIPTION
**Pull request by RonRademaker**
This closes #66 

I'm experiencing lots of timeout problems with some microservices due to the services building up some kind of cache at some requests. This allows me to implement a statusCodeHandler which can detect a timeout and deal with, for example putting in some instruction to load the fragment later using XHR.

Example usage (not suitable for production ;)):

``` json
statusCodeHandlers":{
        "500":{
            "fn":"handleInternalServerError"
        }
    }
```

``` javascript
config.functions = {
    'handleInternalServerError': function(req, res, templateVars, data, transformedOptions, err) {
        if (err.message.match(/ETIMEDOUT$/)) {
            return '<script>document.location.reload();</script>';
        }
    }
};
```

### CODE REVIEW

**Comment by gabceb** [comment]
I'm not the biggest fan of functions returning sync and async because they open the code to future bugs and makes the code hard to follow. Why can't you just hijack the response that gets returned on the async version and make the `response.content` equal what you want the script to be?

**Comment by cliftonc** [comment]
@RonRademaker I've submitted an alternate PR with a slightly different approach thats a bit more inline with the rest of the project (felt simpler than nit-picking on yours).  I think this achieves what you wanted?  #77

**Comment by cliftonc** [comment]
Closed in favour of #77.
